packages/contracts/scripts/write-version.ts

```diff
@@ -1,12 +1,21 @@
-// eslint-disable-next-line @typescript-eslint/no-var-requires
-import { version } from '../package.json'
+import { execSync } from 'child_process'
+
+const getCurrentCommitHash = () => {
+  try {
+    const commitHash = execSync('git rev-parse HEAD').toString().trim()
+    return commitHash
+  } catch (error) {
+    console.error('Error getting current commit hash:', error)
+    return null
+  }
```

Comment:
Optional:

We already have a `getCurrentGitCommitHash` function in the core package (in `utils.ts`). It'd be better to move it to the contracts package instead of essentially duplicating it here. If you do this, you'll also need to move its tests from the core package (in `utils.spec.ts`) to the contracts package. Lastly, since the `getCurrentGitCommitHash` may return `null`, we should throw an error in this file if it returns `null` instead of a string.

Reply:
To clarify: we should check for the null case regardless of whether or not you merge these functions because they can both return null